### PR TITLE
Feature/condo/sberdoma 112/property list

### PR DESCRIPTION
**Pull request by dkoviazin**
1. Made domain folder for the property with code generation
2. 2 virtual fields for the propery model
3. Updated empty container to work not only with tasks
4. 2Gis map view mode for all properties
5. Table view mode with sorting and filtering on address field
list1
list_filter
list_map
list_map_choose

### CODE REVIEW

File: apps/condo/domains/property/gql.js [hunk 1]
```diff
@@ -1,8 +1,29 @@
...
-const Property = genTestGQLUtils('Property', PROPERTY_FIELDS)
+const { generateGqlQueries } = require('@condo/domains/common/utils/codegeneration/generate.gql')
+const gql = require('graphql-tag')
+
+const COMMON_FIELDS = 'id dv sender v deletedAt organization { id name} newId createdBy { id name } updatedBy { id name } createdAt updatedAt'
+
+const PROPERTY_FIELDS = `{ organization { id name} name address addressMeta type ticketsInWork flatsCount map ${COMMON_FIELDS} }`
+const Property = generateGqlQueries('Property', PROPERTY_FIELDS)
+
+const GET_TICKET_INWORK_COUNT_BY_PROPERTY_ID_QUERY = gql`
```
Comment (Dimitreee) [inline apps/condo/domains/property/gql.js:15]: We can use default `Ticket.useObjects()` with filter params.
see: apps/condo/pages/ticket/index.tsx:47

File: apps/condo/domains/property/schema/Property.js [hunk 2]
```diff
@@ -77,9 +88,50 @@ const Property = new GQLListSchema('Property', {
                 },
             },
         },
+        flatsCount: {
```
Comment (Dimitreee) [inline apps/condo/domains/property/schema/Property.js:91]: Add tests please

File: apps/condo/domains/property/schema/Property.js [hunk 3]
```diff
@@ -77,9 +88,50 @@ const Property = new GQLListSchema('Property', {
...
+                            .reduce((acc, current) => acc + current, 0)
+                    } catch (e) {
+                        // [TODO] zuch - rewrite to PropertyUnit count
+                        console.error('Error while fetching virtual field flatsCount', e)
+                    }
+                }
+                return count
+            },
+        },
+        ticketsInWork: {
```
Comment (Dimitreee) [inline apps/condo/domains/property/schema/Property.js:110]: add tests

File: apps/condo/domains/property/utils/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    address?: string
+}
+
+export const getFiltersFromQuery = (query: ParsedUrlQuery): IFilters => {
+    const filters = get(query, 'filters')
+    if (!filters) {
+        return {}
+    }
+    try {
+        // @ts-ignore
```
Comment (Dimitreee) [inline apps/condo/domains/property/utils/helpers.ts:29]: ??

File: apps/condo/domains/property/utils/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        return JSON.parse(filters)
+    } catch (e) {
+        return {}
+    }
+}
+
+export const getSortStringFromQuery = (query: ParsedUrlQuery): Array<string> => {
```
Comment (Dimitreee) [inline apps/condo/domains/property/utils/helpers.ts:36]: add tests for this utils please

File: apps/condo/lang/ru.json [hunk 6]
```diff
@@ -7,12 +7,19 @@
   "pages.condo.e-voting.id.PageTitle": "Онлайн собрание",
   "pages.condo.e-voting.status.draft": "Черновик",
   "pages.condo.e-voting.status.published": "Опубликовано",
-  "pages.condo.property.index.PageTitle": "Объекты (здания, поселки)",
-  "pages.condo.property.index.CreatePropertyButtonLabel": "Создать",
+  "pages.condo.property.index.PageTitle": "Все дома",
```
Comment (Dimitreee) [inline apps/condo/lang/ru.json:10]: add missed fields to en.json

File: apps/condo/pages/property/index.tsx [hunk 7]
```diff
@@ -81,15 +105,20 @@ function CreateAndEditPropertyModalForm ({ action, visible, editableItem, cancel
         </BaseModalForm>
     )
 }
-
+// [todo] zuch - chenge from modal to create page
```
Comment (Dimitreee) [inline apps/condo/pages/property/index.tsx:108]: Comment format is `TODO(username):source`

File: apps/condo/domains/common/components/EmptyListView.tsx [hunk 8]
```diff
@@ -4,37 +4,50 @@ import { useRouter } from 'next/router'
...
-export const EmptyListView = () => {
+
+export interface IEmptyListProps {
+    title: string,
+    text: string,
+    createRoute: string,
+    createLabel: string
+}
+
+export const EmptyListView: React.FC<IEmptyListProps> = ({ title, text, createRoute, createLabel }) => {
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/EmptyListView.tsx:24]: Looks good  👍  👍  👍

File: apps/condo/domains/common/components/MapGL.tsx [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+    text?: string,
+    location: { lat: number, lng: number },
+    route?: string
+}
+
+export interface IMapProps {
+    points: [IPointsOfInterest],
+}
+
+const MapWrapper = React.memo(function MapWrapper () {
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/MapGL.tsx:21]: Why should we use React.memo here?

File: apps/condo/domains/common/components/MapGL.tsx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+        map.setCenter(center)
+        if (points.length) {
+            for (const point of points) {
+                const marker = new api.Marker(map, {
+                    coordinates: _toArrCoordinates(point.location),
+                    icon: MARKER_SVG_URL,                    
+                })
+                marker.on('click', function (e){
+                    setSelected(point)
+                    console.log('Marker is clicked', point)
```
Comment (Dimitreee) [inline apps/condo/domains/common/components/MapGL.tsx:64]: remove debug

File: apps/condo/domains/common/constants/map.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+
+// Key from examples of using 2gis mapgl
+const GUEST_API_KEY = 'bfd8bbca-8abf-11ea-b033-5fa57aae2de7'
```
Comment (Dimitreee) [inline apps/condo/domains/common/constants/map.js:3]: move to .env

File: apps/condo/pages/property/index.tsx [hunk 12]
```diff
@@ -98,75 +127,218 @@ function CreatePropertyModalBlock ({ modal, create }) {
...
+const PropertyPageViewMap = (): React.FC => {
+    const {
+        objs: properties,
+    } = Property.useObjects()
 
-function ViewOrEditPropertyListBlock ({ loading, objs }) {
-    const intl = useIntl()
-    const SelectMsg = intl.formatMessage({ id: 'Select' })
+    const points = properties
+        .filter(property => Reflect.has(property.addressMeta, 'data'))
```
Comment (Dimitreee) [inline apps/condo/pages/property/index.tsx:136]: Is it working? did not see import of Reflect

File: apps/condo/domains/common/components/MapGL.tsx (hunk 10 above)
Comment (dkoviazin) [reply apps/condo/domains/common/components/MapGL.tsx:64]: Ok

File: apps/condo/domains/property/gql.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/property/gql.js:15]: This query is used in virtual field to get count, so I can't use useObjects here

File: apps/condo/domains/property/schema/Property.js (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/domains/property/schema/Property.js:91]: Done

**Pull request closed without merging**